core/domain/user_jobs_one_off_test.py

```diff
@@ -196,6 +196,141 @@ def test_user_contributions_get_created_after_running_the_job(self):
+
+    def _run_one_off_job(self):
+        """Runs the one-off MapReduce job."""
+        job_id = user_jobs_one_off.PopulateUserAuthModelOneOffJob.create_new()
+        user_jobs_one_off.PopulateUserAuthModelOneOffJob.enqueue(job_id)
+        self.assertEqual(
+            self.count_jobs_in_taskqueue(
+                taskqueue_services.QUEUE_NAME_ONE_OFF_JOBS), 1)
+        self.process_and_flush_pending_tasks()
+
+        stringified_output = (
+            user_jobs_one_off.PopulateUserAuthModelOneOffJob.get_output(
+                job_id))
+        output = {}
+        for stringified_distribution in stringified_output:
+            message_list = ast.literal_eval(stringified_distribution)
+            output[message_list[0]] = sorted(message_list[1])
+        return output
+
+    def setUp(self):
+        super(PopulateUserAuthModelOneOffJobTests, self).setUp()
+        self.user_a_model = user_models.UserSettingsModel(
+            id=self.USER_A_ID,
+            gae_id=self.USER_A_GAE_ID,
+            email=self.USER_A_EMAIL,
+        )
+        self.user_b_model = user_models.UserSettingsModel(
+            id=self.USER_B_ID,
+            gae_id=self.USER_B_GAE_ID,
+            email=self.USER_B_EMAIL
+        )
+        self.user_a_model.put()
+
+    def test_before_migration_user_does_not_exist_in_user_auth_model(self):
+        self.assertIsNone(user_models.UserAuthModel.get_by_id(self.USER_A_ID))
+
+    def test_one_off_job_migrate_single_new_user_successfully(self):
+        self.assertIsNone(user_models.UserAuthModel.get_by_id(self.USER_A_ID))
+
+        output = self._run_one_off_job()
+
+        user_auth_model = user_models.UserAuthModel.get_by_auth_id(
+            feconf.AUTH_METHOD_GAE, self.USER_A_GAE_ID)
+        self.assertEqual(user_auth_model.id, self.USER_A_ID)
+
+        expected_output = {
+            'Created': [
+                {'user_id': self.USER_A_ID, 'gae_id': self.USER_A_GAE_ID}]
+        }
+        self.assertEqual(output, expected_output)
+
+    def test_one_off_job_migrate_multiple_new_users_successfully(self):
+        self.user_b_model.put()
+        self.assertIsNone(user_models.UserAuthModel.get_by_id(self.USER_A_ID))
+
+        output = self._run_one_off_job()
+
+        user_auth_model = user_models.UserAuthModel.get_by_auth_id(
+            feconf.AUTH_METHOD_GAE, self.USER_A_GAE_ID)
+        self.assertEqual(user_auth_model.id, self.USER_A_ID)
+        user_auth_model = user_models.UserAuthModel.get_by_auth_id(
+            feconf.AUTH_METHOD_GAE, self.USER_B_GAE_ID)
+        self.assertEqual(user_auth_model.id, self.USER_B_ID)
+
+        expected_output = {
+            'Created': [
+                {'user_id': self.USER_A_ID, 'gae_id': self.USER_A_GAE_ID},
+                {'user_id': self.USER_B_ID, 'gae_id': self.USER_B_GAE_ID}
+            ]
+        }
+        self.assertEqual(expected_output, output)
+
+    def test_one_off_job_existing_user_updated_successfully(self):
```

Comment:
Do we need to add a test where an existing user does not get updated if they already have a gae_id/user_id combination in UserAuthModel ?

Reply:
Not required now. Please see the updated migration job.